plasmapy/formulary/shocks.py

```diff
@@ -0,0 +1,103 @@
+""" This module gathers functions relating to shocks and the properties thereof.
+"""
+__all__ = ["entropy_across_shock_polytropic", "ds_"]
+
+
+import warnings
+from astropy import units as u
+from plasmapy.utils.exceptions import PhysicsWarning
+from plasmapy.utils.decorators import (
+    angular_freq_to_hz,
+    check_relativistic,
+    validate_quantities,
+)
+from astropy.constants import c, a0, k_B
+from numpy import pi, exp, sqrt, log, nan, isclose
+
+
+@validate_quantities(
+    rho_1={"can_be_negative": False},
+    rho_2={"can_be_negative": False},
+    c_v={"can_be_negative": False},
+)
+def entropy_across_shock_polytropic(
+    c_v: u.J / u.K,
+    p_1: u.Pa,
+    p_2: u.Pa,
+    rho_1: u.kg / u.m ** 3,
+    rho_2: u.kg / u.m ** 3,
+    gamma=5 / 3,
+) -> u.J / u.K:
+    r"""
+    Entropy is not conserved across a shock, since a shock is
+    an irreversible, non-adiabatic process. This equation is based on the specific entropy
+    of a polytropic gas before and after the shock has passed.
+
+     .. math::
+
+        s_2 - s_1 =
+        c_v ln\left[ \frac{p_2}{p_1} \left( \frac{\rho_1}{\rho_2} \right)^{\gamma} \right]
+    Here variables indexed by 1 and 2 are upstream (pre shock)
+    and downstream (post shock) respectively.
```

Comment:
If I remember correctly, this is derived from a non-magnetic, two-fluid approach.  What I'm trying to get at, you should put a little backstory into the docstring so users know the equations assumptions and "valid" conditions.

Also, don't forget to cite reference material.